### PR TITLE
cephadm: introduce flake8

### PR DESCRIPTION
**Pull request by mgfritch**
Adds flake8 as a linter for bin/cephadm

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 if(WITH_TESTS)
   include(AddCephTest)
-  add_tox_test(cephadm TOX_ENVS py3 mypy)
+  add_tox_test(cephadm TOX_ENVS py3 mypy, flake8)
```
Comment (liewegas) [inline src/cephadm/CMakeLists.txt:3]: that comma doesn't look right...

File: src/cephadm/cephadm [hunk 2]
```diff
@@ -91,15 +43,62 @@ from threading import Thread, RLock
 from urllib.error import HTTPError
 from urllib.request import urlopen
 
-cached_stdin = None
 
+# Default container images -----------------------------------------------------
+DEFAULT_IMAGE = 'docker.io/ceph/daemon-base:latest-master-devel'
+DEFAULT_IMAGE_IS_MASTER = True
```
Comment (liewegas) [inline src/cephadm/cephadm:49]: worth standardizing on ``'`` or ``"``?

File: src/cephadm/CMakeLists.txt (hunk 1 above)
Comment (mgfritch) [reply src/cephadm/CMakeLists.txt]: oops, fixed!

File: src/cephadm/cephadm (hunk 2 above)
Comment (mgfritch) [reply src/cephadm/cephadm:49]: I'm in favor of that, personally I prefer `'` ..

Comment (mgfritch) [reply src/cephadm/cephadm:49]: @liewegas https://github.com/ceph/ceph/pull/39699

**Comment by mgfritch** [comment]
jenkins test make check

**Comment by mgfritch** [comment]
jenkins test make check

**Comment by mgfritch** [comment]
jenkins test make check

**Comment by jmolmo** [comment]
It seems that this pr can be superseded by https://github.com/ceph/ceph/pull/39699. I am wrong?

**Comment by mgfritch** [comment]
> It seems that this pr can be superseded by #39699. I am wrong?

closing in favor of using a single PR (#39699)

**Pull request closed without merging**